crates/api_crud/src/community/read.rs

```diff
@@ -75,8 +76,23 @@ impl PerformCrud for GetCommunity {
       .await
       .unwrap_or(1);
 
+    let mut site_id: Url = community_view.community.actor_id.clone().into();
+    site_id.set_path("");
+    let mut site: Option<Site> = blocking(context.pool(), move |conn| {
+      Site::read_from_apub_id(conn, site_id)
+    })
+    .await??;
+    // no need to include metadata for local site (its already available through other endpoints).
+    // this also prevents us from leaking the federation private key.
```

Comment:
This might be really confusing for front end clients, which might expect that to be there. (Still don't know what the exact use case is for including this is). 

Might just be easier to set the private key to None, similar to how this is done in SiteView::read_local.

Reply:
I disagree. Doing it like this makes it easier to show the instance metadata only for remote instances, otherwise you would have to add extra logic for that on the frontend. And there are plenty of nullable fields in the api, one more wont make any difference.